src/tracks/InstrumentTrack.cpp

```diff
@@ -859,10 +859,8 @@ void InstrumentTrack::saveTrackSpecificSettings( QDomDocument& doc, QDomElement
 	m_arpeggio.saveState( doc, thisElement );
 
 	// Save the midi port info if we are not in song saving mode, e.g. in
-	// track cloning mode or if we are in song saving mode and the user
-	// has chosen to discard the MIDI connections.
-	if (!Engine::getSong()->isSavingProject() ||
-	    !Engine::getSong()->getSaveOptions().discardMIDIConnections.value())
+	// track cloning mode or if we are in song saving mode
+	if (!Engine::getSong()->isSavingProject())
```

Comment:
I think this statement will be always false, effectively the if block will never be executed. I'll remove it.

Reply:
What do you mean here?

Reply:
@Rossmaxx, the code block is relevant for a bug fix that was reported in https://github.com/LMMS/lmms/issues/5706. See pull request https://github.com/LMMS/lmms/pull/7066.

By the way, oftentimes one can use the blame feature to find out why some code was added. If one is not sure why some code is there it can be very useful to check this.

Reply:
Nope, it was a misconception on my side. I'll just keep the condition like this.